lib/TH/generic/THStorage.c

```diff
@@ -107,9 +107,9 @@ void THStorage_(free)(THStorage *storage)
   if(!storage)
     return;
 
-  if((storage->flag & TH_STORAGE_REFCOUNTED) && (storage->refcount > 0))
+  if((storage->flag & TH_STORAGE_REFCOUNTED) && (THAtomicGet(&storage->refcount) > 0))
   {
```

Comment:
i hesitated to access atomic variables only with atomic functions, as an habit. (in fact, my first version had atomic as a struct, to refrain the user touching directly the number in there). true, here does not do much... i put it only for consistency reasons. you think i should remove it?